### PR TITLE
Vlan trunk common function support

### PR DESCRIPTION
**Pull request by JipanYanga**
Common functions update for VLAN trunk feature

### CODE REVIEW

File: common/gettimestamp.h [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#ifndef __GETTIMESTAMP__
```
Comment (stcheng) [inline common/gettimestamp.h]: rename to `timestamp.h` and use `#pragma once` for all the header files.

File: common/subscriber.cpp [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    m_buffer.pop_front();
+}
+
+/*
+ * handle redis keyspace pmessage notification
+ */
+void Subscriber::pops(std::deque<KeyOpFieldsValuesTuple> &vkco, std::string /*prefix*/)
+{
+    RedisReplyAsync r(getSubscribeDBC());
+    // Keep fetching the reply object until all of them are read
```
Comment (stcheng) [inline common/subscriber.cpp]: align all the comment format to use `/**/`.

File: common/subscriber.cpp [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        ctx = ctx0->element[2];
+        std::string table_key = ctx->str;
+        size_t found = table_key.find(':');
+        // Return if the format of key is wrong
+        if (found == std::string::npos)
+        {
+            SWSS_LOG_ERROR("invalid keyspace notificaton %s", table_key.c_str());
+            continue;
+        }
+        //strip off  the keyspace db string
```
Comment (stcheng) [inline common/subscriber.cpp]: extra space

File: common/subscriber.cpp [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+        auto ctx = ctx0->element[1];
+        if(m_keyspace.compare(ctx->str))
+        {
+            SWSS_LOG_ERROR("invalid pattern %s returned for pmessage of %s", ctx->str, m_keyspace.c_str());
+            continue;
+        }
+
+        ctx = ctx0->element[2];
+        std::string table_key = ctx->str;
+        size_t found = table_key.find(':');
```
Comment (stcheng) [inline common/subscriber.cpp]: the `tokenize` function could be used here to check the size of the returned vector.

File: common/gettimestamp.h (hunk 1 above)
Comment (JipanYanga) [reply common/gettimestamp.h]: Ok.

File: common/subscriber.cpp (hunk 3 above)
Comment (JipanYanga) [reply common/subscriber.cpp]: Ok, will remove it.

File: common/subscriber.cpp (hunk 4 above)
Comment (JipanYanga) [reply common/subscriber.cpp]: Ok, will use tokennize.

File: common/subscriber.cpp (hunk 2 above)
Comment (JipanYanga) [reply common/subscriber.cpp]: Ok, will switch to /* */ for all comments.

**Comment by JipanYanga** [comment]
Thanks for the review!
As suggested,  plan to split it into three new pull request.
<1> exec and timestamp changes.
<2> asynchronous redis subscription channel functions for keyspace notification.
<3> non-blocking netlink socket connection update.
Hope this will make it easier to review .  Also will add more detailed comments for each new PR.
Thanks,
Jipan

**Comment by stcheng** [comment]
@jipanyanga Thanks so much! By splitting the commits, the Git history will make more sense for readability. :)

**Comment by JipanYanga** [comment]
target branch not active.

**Pull request closed without merging**